### DISCUSSION TITLE
Getting Issue While Creating Batch For one Day

### DISCUSSION THREAD

**Discussion by mahajanmahesh935** [Sunbird-Lern/Community#199] [Query]
Hello Team,
@AmiableAnil @Jayaprakash8887 @rhwarrier 
**Background**
We have set up the event management in Sunbird version 4.10.0. As per the flow, users can join the event using the enroll API.
**Issue Details**
We are currently facing an issue when creating a batch for a one-day event using the following CURL request:
**Request URL:**
`{{host}}/learner/course/v1/batch/create`
**Request Method:**
`POST`
**Request Body:**
```
{
    "request": {
        "courseId": "{{courseId}}",
        "name": "{{batchName}}",
        "description": "{{batchDescription}}",
        "enrollmentType": "invite-only",
        "startDate": "2024-06-25",
        "endDate": "2024-06-25",
        "createdBy": "{{userId}}"
    }
}
```
**Response:**
```
{
    "id": "api.course.batch.create",
    "ver": "v1",
    "ts": "2024-06-25 11:34:58:379+0000",
    "params": {
        "resmsgid": null,
        "msgid": "6d59e460-1f97-40b1-8ec7-97f2769bf657",
        "err": "END_DATE_ERROR",
        "status": "END_DATE_ERROR",
        "errmsg": "End date should be greater than start date."
    },
    "responseCode": "CLIENT_ERROR",
    "result": {}
}
```
**Issue Explanation**
The API currently requires the end date to be greater than the start date, making it challenging to create an event that starts and ends on the same day.
**Request for Guidance**
Could you please guide us on how to address this issue? We need to create one-day event batches, and the current API restriction is preventing us from doing so.
Thank you for your assistance. We are looking forward to your guidance to resolve this issue.
cc [@komalm](https://github.com/komalm) [@paritshivani](https://github.com/paritshivani)

**Comment by mahajanmahesh935** [comment]
Hi team,
Your insights and assistance would be highly appreciated in resolving this issue. Looking forward to your response.

Thank you.

**Comment by amitpriyadarshi** [comment]
@mahajanmahesh935 as of date the batch creation api has a validation of start date and end date, where start date should not be greater than or equal to end date. And also it is not configuration driven. You need to put end date at least one day after start date.

cc: @vinukumar-vs @AmiableAnil

**Comment by mahajanmahesh935** [comment]
Hello @amitpriyadarshi Thank you for you Insights,
Our requirement is to create a one-day event that includes a participation certificate, which can be attached to the batch. To achieve this, we also need to create a batch that lasts only one day. Please suggest if there is any other way to accomplish this approach.

**Reply by amitpriyadarshi** [reply]
@mahajanmahesh935 as of now there is no way. You have to make it at least 2 days.
